tests/drivers/clock_control/clock_control_api/src/test_clock_control.c

```diff
@@ -14,97 +14,117 @@ LOG_MODULE_REGISTER(test);
 
 struct device_data {
 	const char *name;
-	u32_t startup_us;
+	u32_t subsys_cnt;
```

Comment:
Is it portable to assume the subsystem range is 0 to n-1? I don't think so. For example, on STM32, the subsystem seems to be a pointer to `struct stm32_pclken` instead.

Why not add a definition like this?

```c
struct device_subsys_data {
	clock_control_subsys_t subsys;
	u32_t subsys_startup_us;
};
```

Then device_data can contain a pointer to an  array of device_subsys_data instead of array of startup times.

No worries if I'm talking nonsense.

Reply:
thanks, good suggestion. done